tmt/steps/provision/local.py

```diff
@@ -133,6 +133,34 @@ def pull(
             extend_options: Optional[list[str]] = None) -> None:
         """ Nothing to be done to pull workdir """
 
+    def acquire_log(self, log_name: str) -> Optional[str]:
+        """fetch and return content of a requested log"""
+        if log_name == 'dmesg':
+            log_content = self.execute(Command('dmesg')).stdout
+        if log_content:
+            return log_content
+        return None
+
+    def store_log(self, log_name: str, log_content: str) -> Optional[str]:
+        """save log content and return the path"""
+        if log_name and log_content:
+            self.write(Path(log_name), log_content)
+            return f'{self.workdir}/{log_name}'
+        return None
+
+    def handle_guest_log(self, log_list: Optional[list[str]] = None) -> None:
+        """get log content and save it to the workdir/provision/,
+        list the log dir in guests.yaml"""
+        log_list = log_list if log_list else []
+        for log_name in log_list:
+            log_content = self.acquire_log(log_name)
+            if log_content:
+                log_path = self.store_log(log_name, log_content)
+                if log_path:
+                    self.write(Path('guests.yaml'), f'{ log_name} : {log_path}', 'a')
```

Comment:
We will need a better way how to expose logs. For now, I would leave this place empty, but we will extend the `Guest` class enough to take care of this. Note that `guests.yaml` may contain multiple guests, this simple approach might break the structure or append all logs to the last guest listed in the file.